pkg/patch/internal/internalfakes/fake_formatter.go

```diff
@@ -0,0 +1,133 @@
+	invocationsMutex sync.RWMutex
+}
+
+func (fake *FakeFormatter) MarkdownToHTML(arg1 string, arg2 string) (string, error) {
+	fake.markdownToHTMLMutex.Lock()
+	ret, specificReturn := fake.markdownToHTMLReturnsOnCall[len(fake.markdownToHTMLArgsForCall)]
+	fake.markdownToHTMLArgsForCall = append(fake.markdownToHTMLArgsForCall, struct {
+		arg1 string
+		arg2 string
+	}{arg1, arg2})
+	fake.recordInvocation("MarkdownToHTML", []interface{}{arg1, arg2})
+	fake.markdownToHTMLMutex.Unlock()
+	if fake.MarkdownToHTMLStub != nil {
+		return fake.MarkdownToHTMLStub(arg1, arg2)
+	}
+	if specificReturn {
+		return ret.result1, ret.result2
+	}
+	fakeReturns := fake.markdownToHTMLReturns
+	return fakeReturns.result1, fakeReturns.result2
+}
+
+func (fake *FakeFormatter) MarkdownToHTMLCallCount() int {
+	fake.markdownToHTMLMutex.RLock()
+	defer fake.markdownToHTMLMutex.RUnlock()
+	return len(fake.markdownToHTMLArgsForCall)
+}
+
+func (fake *FakeFormatter) MarkdownToHTMLCalls(stub func(string, string) (string, error)) {
+	fake.markdownToHTMLMutex.Lock()
+	defer fake.markdownToHTMLMutex.Unlock()
+	fake.MarkdownToHTMLStub = stub
+}
+
+func (fake *FakeFormatter) MarkdownToHTMLArgsForCall(i int) (string, string) {
+	fake.markdownToHTMLMutex.RLock()
+	defer fake.markdownToHTMLMutex.RUnlock()
+	argsForCall := fake.markdownToHTMLArgsForCall[i]
+	return argsForCall.arg1, argsForCall.arg2
+}
+
+func (fake *FakeFormatter) MarkdownToHTMLReturns(result1 string, result2 error) {
+	fake.markdownToHTMLMutex.Lock()
+	defer fake.markdownToHTMLMutex.Unlock()
+	fake.MarkdownToHTMLStub = nil
+	fake.markdownToHTMLReturns = struct {
+		result1 string
+		result2 error
+	}{result1, result2}
+}
+
+func (fake *FakeFormatter) MarkdownToHTMLReturnsOnCall(i int, result1 string, result2 error) {
+	fake.markdownToHTMLMutex.Lock()
+	defer fake.markdownToHTMLMutex.Unlock()
+	fake.MarkdownToHTMLStub = nil
+	if fake.markdownToHTMLReturnsOnCall == nil {
+		fake.markdownToHTMLReturnsOnCall = make(map[int]struct {
+			result1 string
+			result2 error
+		})
+	}
+	fake.markdownToHTMLReturnsOnCall[i] = struct {
+		result1 string
+		result2 error
+	}{result1, result2}
+}
+
+func (fake *FakeFormatter) Invocations() map[string][][]interface{} {
+	fake.invocationsMutex.RLock()
+	defer fake.invocationsMutex.RUnlock()
+	fake.markdownToHTMLMutex.RLock()
+	defer fake.markdownToHTMLMutex.RUnlock()
+	copiedInvocations := map[string][][]interface{}{}
+	for key, value := range fake.invocations {
+		copiedInvocations[key] = value
+	}
+	return copiedInvocations
+}
+
+func (fake *FakeFormatter) recordInvocation(key string, args []interface{}) {
+	fake.invocationsMutex.Lock()
+	defer fake.invocationsMutex.Unlock()
+	if fake.invocations == nil {
+		fake.invocations = map[string][][]interface{}{}
+	}
+	if fake.invocations[key] == nil {
+		fake.invocations[key] = [][]interface{}{}
+	}
+	fake.invocations[key] = append(fake.invocations[key], args)
+}
+
+var _ patch.Formatter = new(FakeFormatter)
```

Comment:
it's more common to write the type assertions with `= &FakeFormatter{}`
the compiler may or may not optimize the runtime call here.

Reply:
same as https://github.com/kubernetes/release/pull/1040#discussion_r375186428